### PR TITLE
net: lwm2m: Add SenML CBOR content writer tests

### PR DESCRIPTION
**Pull request by nordic-mare**
This will add unit tests for the raw CBOR content encoder/decoder.

Signed-off-by: Markus Rekdal [markus.rekdal@nordicsemi.no](mailto:markus.rekdal@nordicsemi.no)

### CODE REVIEW

**Comment by nordic-mare** [comment]
@VeijoPesonen @SeppoTakalo

**Review by SeppoTakalo — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
@oyvindronningstad Looks like there is need to do adjustments on the code generation when it comes to the assumed minimum 64-bit integer value.

https://stackoverflow.com/questions/11270084/why-are-there-differing-definitions-of-int64-min-and-why-do-they-behave-differe

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]

**Review by rlubos — APPROVED** [review]

File: tests/net/lib/lwm2m/content_senml_cbor/prj.conf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+CONFIG_NET_TEST=y
+CONFIG_ZTEST=y
+
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_NEWLIB_LIBC=y
+
+CONFIG_LWM2M=y
+CONFIG_LWM2M_VERSION_1_1=y
+CONFIG_LWM2M_RW_CBOR_SUPPORT=y
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply tests/net/lib/lwm2m/content_senml_cbor/prj.conf:11]: @nordic-mare can you fix this please?

[the remark this answers is not in the document]
Comment (carlescufi) [reply tests/net/lib/lwm2m/content_senml_cbor/prj.conf:11]: Let´s do this in a follow-up

**Review by carlescufi — APPROVED** [review]

**Pull request merged**